plutus-metatheory/src/Main.lagda

```diff
@@ -410,16 +432,16 @@ postulate execP : IO Command
 
 {-# COMPILE GHC execP = execP #-}
 
-evalInput : EvalMode → Format → Input → IO (Either ERROR String)
-evalInput U fmt inp = fmap evalProgramNU (parseU fmt inp)
-evalInput m fmt inp = fmap (evalProgramN m) (parse fmt inp)
+evalInput : EvalMode → BudgetMode → Format → Input → IO (Either ERROR String)
+evalInput U bm fmt inp = fmap (evalProgramNU bm) (parseU fmt inp)
+evalInput m _ fmt inp  = fmap (evalProgramN m) (parse fmt inp)
 
 tcInput : Format → Input → IO (Either ERROR String)
 tcInput fmt inp = fmap typeCheckProgramN (parse fmt inp)
 
 main' : Command → IO ⊤
-main' (Eval (EvalOpts inp fmt m)) = do
-  inj₂ s ← evalInput m fmt inp
+main' (Eval (EvalOpts inp fmt m bm)) = do
+  inj₂ s ← evalInput m bm fmt inp
```

Comment:
I think that a couple of lines below here it won't report the tally/costs in the case of failure.  That's just been changed for `uplc` in #5652.  It's not a big deal though.